### PR TITLE
gh-66944: Note that the `contextlib.closing` example is for illustration purposes

### PR DESCRIPTION
**Pull request by scop**
Just stumbled upon the `urlopen` example in `contextlib.closing` docs and later found #66944 open for it, here's a quick crack at at least documenting the unnecessity these days.

* Issue: gh-66944

----
:books: Documentation preview :books:: https://cpython-previews--112198.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/contextlib.rst [hunk 1]
```diff
@@ -182,6 +182,11 @@ Functions and classes provided:
    without needing to explicitly close ``page``.  Even if an error occurs,
    ``page.close()`` will be called when the :keyword:`with` block is exited.
 
+   .. note::
+
+      As of Python 3, :func:`urllib.request.urlopen` return values can be
+      used as context managers directly; there is no need to use ``closing``
+      with them. The example above is just for illustration purposes.
```
Comment (AA-Turner) [inline Doc/library/contextlib.rst]:
```suggestion
   .. note::

      Most types managing resources support the :term:`context manager` protocol,
      which closes *thing* on leaving the :keyword:`with` statment.
      As such, :func:`!closing` is most useful for third party types that don't
      support context managers.
      This example is purely for illustration purposes,
      as :func:`~urllib.request.urlopen` would normally be used in a context manager.
```

File: Doc/library/contextlib.rst [hunk 2]
```diff
@@ -182,6 +182,14 @@ Functions and classes provided:
    without needing to explicitly close ``page``.  Even if an error occurs,
    ``page.close()`` will be called when the :keyword:`with` block is exited.
 
+   .. note::
+
+      Most types managing resources support the :term:`context manager` protocol,
+      which closes *thing* on leaving the :keyword:`with` statment.
```
Comment (danielhollas) [inline Doc/library/contextlib.rst:188]:
```suggestion
      which closes *thing* on leaving the :keyword:`with` statement.
```

**Review by danielhollas — COMMENTED** [review]
Late to the party but spotted a typo...

**Comment by AA-Turner** [comment]
Thanks Daniel, would you want to open a PR? You can add me as a reviewer.

A

**Pull request merged**